did.go

```diff
@@ -126,6 +191,74 @@ var DIDMethodNetwork = map[DIDMethod]map[DIDNetworkFlag]byte{
 	},
 }
 
+// DIDMethodNetworkParams is a structure to represent DID method network options
+type DIDMethodNetworkParams struct {
+	Method      DIDMethod
+	Blockchain  Blockchain
+	Network     NetworkID
+	NetworkFlag byte
+	chainID     *int
```

Comment:
Why are these fields private and pointers?

Reply:
Just in case, it's ok to have chainID=0 when there's no chain linked. And methodByte must always be filled.

Reply:
It's private field for params, because it is possible to set them in using Options. As we do in similar concept in other libraries. It could be another structure but I see no sense.

DID method is optional because it can be already be registered, no need to duplicate info in this case.

See example of using:

```	params := core.DIDMethodNetworkParams{
		Method:      core.DIDMethodIden3,
		Blockchain:  "linea",
		Network:     "testnet",
		NetworkFlag: 0b0100_0000 | 0b0000001,
	}
	err = core.RegisterDIDMethodNetwork(params, core.WithChainID(59140))
```